DisCatSharp.ApplicationCommands/Workers/RegistrationWorker.cs

```diff
@@ -0,0 +1,382 @@
+        private static bool CheckNotAnyOrNull(Dictionary<ulong, DiscordApplicationCommand> dict)
+            => dict == null || !dict.Any() || dict.Keys == null || !dict.Keys.Any();
+
+        private static bool CheckAnyAndNotNull(List<DiscordApplicationCommand> list)
+            => list != null && list.Any();
+
+        private static bool CheckAnyAndNotNull(List<ulong> list)
+            => list != null && list.Any();
+
+        private static bool CheckAnyAndNotNull(Dictionary<ulong, DiscordApplicationCommand> dict)
+            => dict != null && dict.Any() && dict.Keys != null && dict.Keys.Any();
+        #endregion
+
+        /// <summary>
+        /// Builds a list of guild command ids to be deleted on discords backend.
+        /// </summary>
+        /// <param name="guildId">The guild id these commands belong to.</param>
+        /// <param name="updateList">The command list.</param>
+        /// <returns>A list of command ids.</returns>
+        private static List<ulong> BuildGuildDeleteList(ulong guildId, List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._guildDiscordCommands == null || !ApplicationCommandsExtension._guildDiscordCommands.Any()
+                || ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value == null
+            )
+                return null;
+
+            var discord = ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value;
+
+            List<ulong> InvalidCommandIds = new();
+
+            if (updateList == null)
+            {
+                foreach (var cmd in discord)
+                {
+                    InvalidCommandIds.Add(cmd.Id);
+                }
+            }
+            else
+            {
+                foreach (var cmd in discord)
+                {
+                    if (!updateList.Any(ul => ul.Name == cmd.Name))
+                        InvalidCommandIds.Add(cmd.Id);
+                }
+            }
+
+            return InvalidCommandIds;
+        }
+
+        /// <summary>
+        /// Builds a list of guild commands to be created on discords backend.
+        /// </summary>
+        /// <param name="guildId">The guild id these commands belong to.</param>
+        /// <param name="updateList">The command list.</param>
+        /// <returns></returns>
+        private static List<DiscordApplicationCommand> BuildGuildCreateList(ulong guildId, List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._guildDiscordCommands == null || !ApplicationCommandsExtension._guildDiscordCommands.Any()
+                || updateList == null || ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value == null
+            )
+                return updateList;
+
+            var discord = ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value;
```

Comment:
I just realized we have this where clause happening twice, here and line 248. We could do a FirstOrDefault beforehand? Eliminate the same operation being done twice